src/Service/EntityMetadata.php

```diff
@@ -249,18 +208,28 @@ public function extractRelated(ReflectionClass $reflection): array
             $relatedProperties = [];
 
             foreach ($properties as $property) {
-                $related = $this->annotationReader->getPropertyAnnotation(
-                    $property,
-                    Related::class
-                );
-                $exposed = $this->annotationReader->getPropertyAnnotation(
-                    $property,
-                    Expose::class
-                );
-
-                if ($related && $exposed) {
-                    $relatedProperties[$property->getName()] =
-                        $related->value ? $related->value : $property->getName();
+                $related = $property->getAttributes(\App\Attribute\Related::class);
+                $exposed = $property->getAttributes(\App\Attribute\Related::class);
+                if (count($related)) {
```

Comment:
could this be simplified?

```suggestion
                if (count($related) && count($exposed)) {
```

Reply:
It's an edge case that I don't think we actually have, but if there are `Related` things on a class, but none of them are exposed making this change would cause the next block searching for annotations to run instead of being skipped. By splitting into two ifs we only run the annotation block in that case.

Reply:
:heavy_check_mark: